docs/writing_a_simple_publisher_and_subscriber.md

```diff
@@ -0,0 +1,479 @@
+
+To construct a node, replace the code in your `main.rs` file with the following:  
+```rust
+/// Creates a SimplePublisherNode, initializes a node and publisher, and provides
+/// methods to publish a simple "Hello World" message on a loop in separate threads.
+
+use rclrs::{create_node, Context, Node, Publisher, RclrsError, QOS_PROFILE_DEFAULT};
+use std::{env, sync::Arc, thread, time::Duration};
+use std_msgs::msg::String as StringMsg;
+/// SimplePublisherNode struct contains node and publisher members.
+/// Used to initialize a ROS 2 node and publisher, and publish messages.
+struct SimplePublisherNode {
+    node: Arc<Node>,
+    publisher: Arc<Publisher<StringMsg>>,
+}
+impl SimplePublisherNode {
+    fn new(context: &context) -> result<self, RclrsError> {
+        let node = create_node(context, "simple_publisher").unwrap();
+        let publisher = node
+            .create_publisher("publish_hello", qos_profile_default)
+            .unwrap();
+        ok(self { node, publisher })
+    }
+    fn publish_data(&self, increment: i32) -> Result<i32, RclrsError> {
+        let msg: StringMsg = StringMsg {
+            data: format!("Hello World {}", increment),
+        };
+        self.publisher.publish(msg).unwrap();
+        Ok(increment + 1_i32)
+    }
+}
+fn main() -> Result<(), RclrsError> {
+    let context = Context::new(env::args()).unwrap();
+    let publisher = Arc::new(SimplePublisherNode::new(&context).unwrap());
+    let publisher_other_thread = Arc::clone(&publisher);
+    let mut count: i32 = 0;
+    thread::spawn(move || loop {
+        thread::sleep(Duration::from_millis(1000));
+        count = publisher_other_thread.publish_data(count).unwrap();
+    });
+    rclrs::spin(publisher.node.clone())
+}
+```
+
+<details><summary>Examining the code in detail:</summary>
+
+#### Importing 
+The first 3 lines of the Rust code imports tools for thread synchronization, time 
+handling, iteration, threading, ROS 2 communication, and string message publishing.
+```rust
+use rclrs::{create_node, Context, Node, Publisher, RclrsError, QOS_PROFILE_DEFAULT};
+use std::{env, sync::Arc, thread, time::Duration};
+use std_msgs::msg::String as StringMsg;
+```
+* `use std::{sync::Arc, time::Duration, iter, thread};`: Imports specific features from the standard library: 
+    - `Arc` is for thread-safe shared ownership of data. 
+    - `Duration` represents a time span. 
+    - `iter` provides tools for working with iterators.
+    - `thread` enables creating and managing threads.
+* `use rclrs::{RclrsError, QOS_PROFILE_DEFAULT, Context, create_node, Node, Publisher};`: 
+    - Imports elements for ROS 2 communication: 
+        - `RclrsError` for handling errors. 
+        - `QOS_PROFILE_DEFAULT` for default Quality of Service settings. 
+- `Context, create_node, Node, Publisher` are for ROS 2 node creation and publishing.
+* `use std_msgs::msg::String as StringMsg;`: Imports the `StringMsg` type for publishing string messages.  
```

Comment:
The order of the imports in the example code does not match the order of the bullet points where we expand whats going on in each line.